usaspending_api/etl/elasticsearch_loader_helpers/delete_data.py

```diff
@@ -352,13 +392,26 @@ def delete_transactions(
 
     Returns: Number of ES docs deleted in the index
     """
+
     deleted_tx_keys = _gather_deleted_transaction_keys(
         config, fabs_external_data_load_date_key, fpds_external_data_load_date_key
     )
+    pre_fy2008_transactions = _gather_modified_transactions_pre_fy2008(config, spark)
+    pre_fy2008_transactions_len = len(pre_fy2008_transactions)
+
+    if deleted_tx_keys is not None and pre_fy2008_transactions_len > 0:
+        tx_keys_to_delete = list(set([*deleted_tx_keys] + pre_fy2008_transactions))
+    elif deleted_tx_keys is None and pre_fy2008_transactions_len > 0:
+        tx_keys_to_delete = pre_fy2008_transactions
+    elif deleted_tx_keys is not None and pre_fy2008_transactions_len == 0:
+        tx_keys_to_delete = [*deleted_tx_keys]
+    else:
+        tx_keys_to_delete = []
```

Comment:
This block of conditionals seems clunky to me. Could this be simplified? I know the datastructures are set up a little different (so the same exact approach can't be used), but the `delete_awards()` approach of combining the two potential sets of deleted awards seems much cleaner
```
    values_list = list(
        set(
            [v for d in deleted_award_kvs for v in d.values()]
            + [v for d in updated_awards_pre_fy2008 for v in d.values()]
        )
    )
```